graphql/resolvers.js

```diff
@@ -293,13 +306,68 @@ const updatePost = async ({ id, postInput }, req) => {
   return responseData;
 };
 
+const deletePost = async ({ id }, req) => {
+  try {
+    //  Check if user not authenticated
+    if (!req.isAuth) {
+      const error = new Error('Not authenticated');
+      error.code = 401;
+      throw error;
+    }
+
+    // Get post ID
+    const getPost = await Post.findById(id);
+
+    // If post doesn't exist
+    if (!getPost) {
+      const error = new Error('No post found');
+      error.code = 404;
+      throw error;
+    }
+
+    if (getPost.creator.toString() !== req.userId.toString()) {
+      const error = new Error('Not authorized');
+      error.code = 403;
+      throw error;
+    }
+
+    // Delete image that belog to post
+    clearImage(getPost.imageUrl);
+
+    //  Remove post by id
+    await Post.findByIdAndRemove(id);
+
+    //  Validate user post has removed
+    const getUser = await User.findById(req.userId);
+
+    console.log('Pull the post: ', getUser.posts.pull(id));
+
+    //  Acces post that just deleted by id and Check if a post still exist, Delete post failure
+    if (getUser.posts.pull(id).length !== 0) {
+      return false;
+    }
```

Comment:
Make sure that post is deleted by pulling the post by id if an array doesn't empty that's means deleting post failure.